book/quick-brick/tvOS/Plugins/PlayerDependants/intro.md

```diff
@@ -0,0 +1,120 @@
+## TvOS Video Player Depandant plugins
+
+### Introduction
+This Document will explain general structure how Video Player Depandant plugin are working on TvOS with QuickBrick
+
+#### Supported Apple TV variant:
+Apple TV 4th Generation or later using tvOS 11.0 or higher
+
+#### Available from ZappSDK 10.1.1 and above
+
+#### Required Applicaster Frameworks:
+
+* ZappPlugins
+
+### Content
+* <a href="#general">General</a>
+* <a href="#api">API</a>
+* [How To Create New Plugin?](/quick-brick/tvOS/Plugins/Video/VideoPluginsHowCreate.md)
+
+### Plugins
+
+* [Google Ineractive media Ads](/quick-brick/tvOS/Plugins/Video/DefaultPlayer/DefaultPlayer.md)
+
+### Dependant Player types
+* `advertismentPlaying` - Video ads plugins type
+
+<a name="general" />
+
+### General
+
+Video Player Depandant plugins, plugin type that require player instance to the invocation. Example of such plugins can be GoogleIMA or Analytics for the player.
+
+To implement type of such plugins need implemented on player plugin side two protocols:
+`QBPlayerProtocol` - This protocol define what var and what func can be retrieved from the player
+`ZPDependablePlayerPluginProtocol` - Implementation of this protocol will defined what types of dependant plugin player will support
+
+On Player dependant plugins needed to be implement two protocols
+`ZPPlayerDependantPluginProtocol` - This protocol defines that plugin can get player instance to retieve data
+`QBPlayerObserverProtocol` - By this protocol player will send event to plugin that dependant by player
+
+Between Player plugin and dependant plugins is `ZPPlayerDependantPluginsManager:QBPlayerObserverProtocol`. It is create dependant plugins when player intialized and remove them when player finished to work. Also it resends events to plugins that conform protocol `QBPlayerObserverProtocol`
+
+Behind the scenes the logic is next:
+1. Player send to `ZPPlayerDependantPluginsManager` its life cycles events on method `playerDidCreate:`
+Manager will create instances of dependant plugins if player support current plugins types defined in `ZPDependablePlayerPluginProtocol`
+2. Manager will send events to dependant plugins.
+3. When Player will finish work and send event `playerDidFinishPlayItem:` manager will wait completion from plugins in case something needed to do before next video will start or close player. Can be good timing for presentation postroll ad as example.
+4. When Player will call `playerDidDismiss:`  Manager will release all dependants  plugins availible for current playter.
+
+<a name="protocols" />
+#### Protocols
+
+###### QBPlayerProtocol
+
+	// DS raw atom entry format
```

Comment:
ALL this code needs to be within a block
```swift
  (code here)
```
With the explanation outside the block of code